SignalServiceKit/src/Contacts/Contact.m

```diff
@@ -127,12 +129,26 @@ - (instancetype)initWithSystemContact:(CNContact *)contact
     _emails = [emailAddresses copy];
 
     if (contact.thumbnailImageData) {
-        _image = [UIImage imageWithData:contact.thumbnailImageData];
+        _imageData = contact.thumbnailImageData;
     }
 
     return self;
 }
 
+- (nullable UIImage *)image
+{
```

Comment:
For equality checks, it's easier to compare the underlying image data than an instantiated UIImage, so I've shuffled things around to make the Data the persisted property, and the UIImage generated lazily from it.

Reply:
Comparing image-sized `NSData` instances will be expensive; we should make sure we never do it on the main thread.

Reply:
They are thumbnails, so they don't get too large (a couple dozen kb). Admittedly it's not free, but doing this check, even when on the main thread, is far cheaper than the consequence of firing the unnecessary notification.